packages/gatsby/src/utils/cache.js

```diff
@@ -1,73 +1,47 @@
-const Promise = require(`bluebird`)
 const fs = require(`fs-extra`)
-const _ = require(`lodash`)
+const manager = require(`cache-manager`)
+const fsStore = require(`cache-manager-fs`)
+const path = require(`path`)
 
-const objectToMap = obj => new Map(Object.entries(obj))
+const MAX_CACHE_SIZE = 250
+const ONE_HOUR = 60 * 60
 
-const mapToObject = map => {
-  const obj = {}
-  for (let [key, value] of map) {
-    obj[key] = value
+class Cache {
+  constructor(folderName = `db`, store = fsStore) {
+    this.folderName = folderName
+    this.store = store
   }
-  return obj
-}
 
-let db
-let directory
-let save
+  get directory() {
+    return path.join(process.cwd(), `.cache/caches/${this.folderName}`)
+  }
+
+  init() {
+    fs.ensureDirSync(this.directory)
+
+    const caches = [
+      { store: `memory`, max: MAX_CACHE_SIZE },
+      {
+        store: this.store,
+        options: {
+          path: this.directory,
+          ttl: ONE_HOUR,
```

Comment:
I am not sure if this is too high or too low 🤷‍♂️

Reply:
I think TTL needs to be tested - I'm not sure if we actually want this to ever expire, but from our brief testing it seemed like TTL didn't actually do anything here

Reply:
Yeah, so our old mechanism _never_ expired because there wasn't any TTL baked in. We could do something reasonable (like one hour?) or go a little higher... maybe 24 hours?

Reply:
this is FS store setting, right? Why should it actually expire? This will make subsequent builds (depending on where cache is used) potentially longer. For remark plugin I think it's reasonable - as build artefacts are cached as query results so caching internal data is potentially waste of disk space, but there are other plugins that use cache - i.e. wordpress plugin to know if it needs to try to redownload images or `sqip` plugin that holds results of cpu intensive tasks in cache, so even if queries need to rerun it will use cached values for `sqip` part of query results

Reply:
Yup. I suppose disk space is the best rationale (although I'm not sure that's a huge concern). But the key itself will be the main means of expiration, i.e. when content in the node changes, the key will inherently be new/new file/etc.